static/js/markdown.js

```diff
@@ -130,9 +130,38 @@ export function apply_markdown(message) {
                 full_name = match[1];
                 user_id = Number.parseInt(match[2], 10);
 
-                if (!helpers.is_valid_full_name_and_user_id(full_name, user_id)) {
-                    user_id = undefined;
-                    full_name = undefined;
+                if (full_name === undefined) {
+                    // For @**|id** syntax
+                    if (!helpers.is_valid_user_id(user_id)) {
+                        // silently ignore invalid user id.
+                        user_id = undefined;
+                    } else {
+                        full_name = helpers.get_actual_name_from_user_id(user_id);
+                    }
+                } else {
+                    // For @**user|id** syntax
+                    if (!helpers.is_valid_full_name_and_user_id(full_name, user_id)) {
+                        user_id = undefined;
+                        full_name = undefined;
+                    }
+                }
+            } else {
+                // check for @**|email@example.com**
+
+                // A simple regex to match |email@example.com
+                // syntax. If email not valid we will not get
+                // person corresponding to it so there is no
+                // need to have a pattern for validating emails
+                // here.
+
+                const email_regex = /\|(.+@.+)/g;
```

Comment:
This is also missing anchors `^` and `$`, so you’ll incorrectly match `@**Wrong Full Name|email@example.com**`, and has a needless `g` flag.